pkgs/applications/networking/instant-messengers/arrpc/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, runtimeShell
+, nodejs
+}:
+
+buildNpmPackage {
+  pname = "arrpc";
+  version = "3.2.0";
+
+  src = fetchFromGitHub {
+    owner = "OpenAsar";
+    repo = "arrpc";
+    # Release commits are not tagged
+    # release: 3.2.0
+    rev = "9c3e981437b75606c74ef058c67d1a8c083ce49a";
+    hash = "sha256-tsO58q6tqqXCJLjZmLQGt1VtKsuoqWmh5SlnuDtJafg=";
+  };
+
+  npmDepsHash = "sha256-vxx0w6UjwxIK4cgpivtjNbIgkb4wKG4ijSHdP/FeQZ4=";
+
+  buildPhase = ''
+    runHook preBuildPhase
+    mkdir -p $out/bin
+
+    cat <<-EOF > $out/bin/arrpc
+      #!${runtimeShell}
+      ${nodejs}/bin/node $out/lib/node_modules/arrpc/src
+    EOF
```

Comment:
Can we ask upstream to add "bin" to their package.json?

Reply:
Opened an upstream issue: https://github.com/OpenAsar/arrpc/issues/49

Reply:
How about making a PR and using `fetchpatch` in the expression?

Reply:
Good thinking! Also it looks like they already merged it. Should I bump it to an unstable version and remove the patch or wait until a future release?

Reply:
I think using the patch makes sense.

Reply:
Okay. In that case, I will keep the patch and the current version.